apps/web/src/pages/integrations/components/multi-provider/CreateProviderInstanceSidebar.tsx

```diff
@@ -47,38 +56,44 @@ export function CreateProviderInstanceSidebar({
     ICreateIntegrationBodyDto
   >(createIntegration);
 
-  const { handleSubmit, control, reset } = useForm({
+  const { handleSubmit, control, reset } = useForm<ICreateProviderInstanceForm>({
     shouldUseNativeValidation: false,
     defaultValues: {
       name: '',
       environmentId: '',
     },
   });
 
-  const onCreateIntegrationInstance = async (data) => {
+  const onCreateIntegrationInstance = async (data: ICreateProviderInstanceForm) => {
     try {
       if (!provider) {
         return;
       }
 
+      const { channel: selectedChannel } = provider;
+      const { environmentId } = data;
+      const hasSimilarIntegration = !!integrations?.find(
+        (el) => el.channel === selectedChannel && el._environmentId === environmentId
+      );
```

Comment:
when the integration is created we do calculate if there are already similar integrations for the env and channel and if there are then we do pass the `hasSimilarIntegration` property to the callback, which then displays or not the modal